py/torch_tensorrt/_compile.py

```diff
@@ -88,7 +88,7 @@ def compile(module: Any, ir="default", inputs=[], enabled_precisions=set([_enums
     target_ir = _module_ir(module, ir)
     if target_ir == _IRType.ts:
         ts_mod = module
-        if isinstance(module, torch.nn.Module):
+        if not isinstance(module, torch.jit.ScriptModule):
```

Comment:
This wont necessarily hold for future cases. For example say someone provides a module in FX format

Reply:
I would write a "_parse_module_type" method to solve this problem.